scripts/o3de/o3de/export_project.py

```diff
@@ -15,132 +15,198 @@
 import sys
 
 from o3de import manifest, utils
-logger = logging.getLogger('o3de.export_project')
+from o3de.validation import *
+
+#grab process utilities from Tools/LyTestTools
+sys.path.insert(0, os.path.join(os.path.dirname(__file__),"..", "..","..","Tools","LyTestTools","ly_test_tools","environment"))
+import process_utils as putil
+
+logger = logging.getLogger(__name__)
 logging.basicConfig(format=utils.LOG_FORMAT)
 
+READ_ONLY_PARAMS = ["export_script", "project_path", "engine_path", "logger", "args"]
+LOGGER_PARAM = "logger"
+
+# a singleton which manages all parameters for the duration of the export process
+# the export script and any script it calls via execute_python_script have access to these params 
+class O3DEScriptExportParameters(utils.BaseSingleton):
+    _attributes = {}
+    _read_only_attribute_names = set()
+
+    def __setattr__(self, attr, value):
+        if attr in O3DEScriptExportParameters._read_only_attribute_names:
+            logger.error(f"Cannot set '{attr}' to '{value}', it is a read-only parameter!")
+        else:
+            O3DEScriptExportParameters._attributes[attr] = value
+    def __getattr__(self, attr):
+        if attr in O3DEScriptExportParameters._attributes: 
+            return O3DEScriptExportParameters._attributes[attr]
+        else:
+            logger.error(f"Attribute '{attr}' does not exist in export parameters!")
+            return None
+    def as_dict(self):
+        return O3DEScriptExportParameters._attributes
 
-class o3de_export_parameters(object):
-    pass
 
 #Helper API
+def mark_readonly_params(*params:str):
+    for p in params:
+        O3DEScriptExportParameters._read_only_attribute_names.add(p)
+
+def mark_writeable_params(*params:str):
+    for p in params:
+        if p in O3DEScriptExportParameters._read_only_attribute_names:
+            O3DEScriptExportParameters._read_only_attribute_names.remove(p)
+
 def process_command(args: list,
                     cwd: pathlib.Path = None,
                     env: os._Environ = None,
-                    show_logging: bool = True):
-    process = Popen(args, cwd=cwd, env=env, stdout=PIPE, stderr=PIPE)
-
-    #read any log lines coming from subprocess
-    while process.poll() is None:
-        line = process.stdout.readline()
-        if not line: break
-
-        log_line = line.decode('utf-8')
-        if show_logging:
-            print(log_line)
-        
-        logger.info(log_line)
+                    show_logging: bool = True) -> int:
+    
+    logger = O3DEScriptExportParameters().logger
     
-    #flush out any remaining text from stdout
-    log_lines = process.stdout.read().decode('utf-8')
-    if show_logging:
-        print(log_lines)
+    def poll_process(process):
+        #read any log lines coming from subprocess
+        while process.poll() is None:
+            line = process.stdout.readline()
+            if not line: break
+            log_line = line.decode('utf-8')
```

Comment:
Its possible to have utf-8 non-decodable bytes from stdout, which means if that occurs this will raise a decoding error. Unless the content is absolutely critical to be exact, you can have this ignore decoding errors like
```suggestion
            log_line = line.decode('utf-8', 'ignore')
```